### ISSUE TITLE
Mark all as read feature

### ISSUE THREAD

**Issue by facundofc** [matterhorn-chat/matterhorn#708]
Hi there! For me it would be very useful to have a key combination and/or command to mark all unread messages as read. We use Mattermost at work, and so I'd use this when returning from a weekend or holidays. At such situations the unread messages predate pretty much every channel we have, and it's awfully time consuming to review them all.

So, I fallback to hitting ALT+A repeatedly, which is slow and boring. Having a command to do the equivalent of that would be awesome (I think a command is more appropriate here so as to prevent one from performing that action inadvertently).

Thanks!

**Comment by jtdaugherty** [comment]
Thanks for the request. Would you expect this to work for just the currently-selected team, or all teams? (I would be a bit wary of marking all channels as read for all teams, but I don't know if you're in a ton of teams.)

**Comment by facundofc** [comment]
Hi Jonathan, thanks for considering this request!

We don't use the teams feature in the organization I'm at now, so I wouldn't really know. From what I was able to grasp from quick google searches, I get that the channels' and direct chats' list shows only those belonging to your current team. Hence, you could have a lot of unread messages in channels not currently displayed. If that is really the case, I'd lean towards just marking as read messages on the currently-selected team. I usually prefer shortcuts and commands to act on things I can see on the screen at the moment I input them. This is just my opinion though.

Cheers!

**Comment by jtdaugherty** [comment]
Okay, thanks! That makes sense.

**Comment by kquick** [comment]
Actually, there's a slight wrinkle to the above:  if you have a direct-message channel to a user and that user is a member of both teams, that channel is visible when you are viewing either team.

Would you expect this clear operation to apply to only to group channels, or would you also expect it to clear direct channels, @facundofc ?  I would  think that we might want this to only apply to the former, but what are your thoughts?

**Comment by jtdaugherty** [comment]
Yeah, I was thinking about that, too. I don't think "mark all as read" should skip DM channels, personally. I also don't think it will be weird to make them as read in one team and then see them read in another. The users are server-wide, not team-specific. (Although I forget whether there is a server setting or preference that constrains the listed users to just members of the current team.) My personal preference is for any command or keystroke for this feature to mark all visible channels as read, including DM channels.

**Comment by facundofc** [comment]
(I changed my mind while I was typing my reply :D)

I haven't thought about this till now. I think I'd prefer for unread DMs
not to be cleared.

The rationale is that DMs are usually stuff that I'm very  interested in,
or that I have to read in order to reply or do something. If those were to
be marked as read along the public channels', I'd find myself a bit in the
reverse situation of the one I'm trying to avoid: I'd have to go check open
DMs' chats one by one to see if someone wrote me something.

Clearing unread messages in public channels doesn't pose this problem,
since what I generally do is search for mentions.

I have to say, though, that in my particular case, the number of public
channels' unread *largely* outnumbers the number of DMs.

So my workflow when coming in after a long weekend or similar would be:

1. Open MM: oh shut... look at all those unread...
2. Clear all public channels' unread
3. Search for mentions
4. Checkout those DMs' unread that still remain

Hope this helps clarify.

Thanks a lot for your attention.

**Comment by jtdaugherty** [comment]
Thanks for writing that up, @facundofc!

**Comment by jtdaugherty** [comment]
Reading over this now, it occurs to me that part of the work on this should include investigation of how the official clients behave, if they support this feature at all.

**Comment by NeilHanlon** [comment]
Hiya - I'd love to see this feature too. I don't think the official clients have a 'mark all as read' feature, but they certainly allow you to clear the unread flag on a channel without actually reading the history. This is also especially useful when travelling and with low bandwidth. Ideally, the unread state could be optionally syncronized between matterhorn and other clients.
